cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -64,6 +64,9 @@ func ValidateClusterConfiguration(c *kubeadm.ClusterConfiguration) field.ErrorLi
 	allErrs = append(allErrs, ValidateDNS(&c.DNS, field.NewPath("dns"))...)
 	allErrs = append(allErrs, ValidateNetworking(c, field.NewPath("networking"))...)
 	allErrs = append(allErrs, ValidateAPIServer(&c.APIServer, field.NewPath("apiServer"))...)
+	allErrs = append(allErrs, ValidateControlPlaneComponent(&c.APIServer.ControlPlaneComponent, field.NewPath("apiServer"))...)
```

Comment:
How about merging this into `ValidateAPIServer` ?

Reply:
yes, we could use the dedicated functions.